packages/front-end/components/Experiment/DifferenceTypeChooser.tsx

```diff
@@ -0,0 +1,192 @@
+import clsx from "clsx";
+import { SetStateAction, useCallback, useEffect, useState } from "react";
+import {
+  ExperimentSnapshotAnalysis,
+  ExperimentSnapshotAnalysisSettings,
+  ExperimentSnapshotInterface,
+} from "back-end/types/experiment-snapshot";
+import { getSnapshotAnalysis } from "shared/util";
+import { DifferenceType } from "back-end/types/stats";
+import { FaCheck } from "react-icons/fa";
+import Dropdown from "@/components/Dropdown/Dropdown";
+import { useAuth } from "@/services/auth";
+import track from "@/services/track";
+import LoadingSpinner from "@/components/LoadingSpinner";
+
+export async function analysisUpdate(
```

Comment:
Where should this method live?

Reply:
typically the call and related state changes could be part of a custom hook, but not sure if it's necessary...